packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FGMClusterManagersController.m

```diff
@@ -0,0 +1,137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FGMClusterManagersController.h"
+
+#import "FGMMarkerUserData.h"
+#import "FLTGoogleMapJSONConversions.h"
+
+@interface FGMClusterManagersController ()
+
+/// A dictionary that cluster managers unique identifiers to GMUClusterManager instances.
+@property(strong, nonatomic)
+    NSMutableDictionary<NSString *, GMUClusterManager *> *clusterManagerIdentifierToManagers;
+
+/// The callback handler interface for calls to Flutter.
+@property(strong, nonatomic) FGMMapsCallbackApi *callbackHandler;
+
+/// The current GMSMapView instance on which the cluster managers are operating.
+@property(strong, nonatomic) GMSMapView *mapView;
+
+@end
+
+@implementation FGMClusterManagersController
+- (instancetype)initWithMapView:(GMSMapView *)mapView
+                callbackHandler:(FGMMapsCallbackApi *)callbackHandler {
+  self = [super init];
+  if (self) {
+    _callbackHandler = callbackHandler;
+    _mapView = mapView;
+    _clusterManagerIdentifierToManagers = [[NSMutableDictionary alloc] init];
+  }
+  return self;
+}
+
+- (void)addJSONClusterManagers:(NSArray<NSDictionary *> *)clusterManagersToAdd {
+  for (NSDictionary *clusterDict in clusterManagersToAdd) {
+    NSString *identifier = clusterDict[@"clusterManagerId"];
+    [self addClusterManager:identifier];
+  }
+}
+
+- (void)addClusterManagers:(NSArray<FGMPlatformClusterManager *> *)clusterManagersToAdd {
+  for (FGMPlatformClusterManager *clusterManager in clusterManagersToAdd) {
+    NSString *identifier = clusterManager.identifier;
+    [self addClusterManager:identifier];
+  }
+}
+
+- (void)addClusterManager:(NSString *)identifier {
+  id<GMUClusterAlgorithm> algorithm = [[GMUNonHierarchicalDistanceBasedAlgorithm alloc] init];
+  id<GMUClusterIconGenerator> iconGenerator = [[GMUDefaultClusterIconGenerator alloc] init];
+  id<GMUClusterRenderer> renderer =
+      [[GMUDefaultClusterRenderer alloc] initWithMapView:self.mapView
+                                    clusterIconGenerator:iconGenerator];
+  GMUClusterManager *clusterManager = [[GMUClusterManager alloc] initWithMap:self.mapView
```

Comment:
What's the purpose of this local variable, rather than assigning directly into the map?